packages/wdio-browserstack-service/src/Percy/Percy.ts

```diff
@@ -0,0 +1,178 @@
+import fs from 'node:fs'
+import path from 'node:path'
+import os from 'node:os'
+
+import { createRequire } from 'node:module'
+const require = createRequire(import.meta.url)
+const { spawn } = require('node:child_process')
+
+import { nodeRequest, getBrowserStackUser, getBrowserStackKey } from '../util.js'
+import { PercyLogger } from './PercyLogger.js'
+
+import PercyBinary from './PercyBinary.js'
+
+import type { BrowserstackConfig, UserConfig } from '../types.js'
+import type { Options } from '@wdio/types'
+
+const logDir = 'logs'
+
+class Percy {
+    #logfile: string = path.join(logDir, 'percy.log')
+    #address: string = process.env.PERCY_SERVER_ADDRESS || 'http://127.0.0.1:5338'
+
+    #binaryPath: string | any = null
+    #options: BrowserstackConfig & Options.Testrunner
+    #config: Options.Testrunner
+    #proc: any = null
+    #isApp: boolean = false
+    #projectName: string | undefined = undefined
+
+    isProcessRunning = false
+
+    constructor(options: BrowserstackConfig & Options.Testrunner, config: Options.Testrunner, bsConfig: UserConfig) {
+        this.#options = options
+        this.#config = config
+        if (options.app) {
+            this.#isApp = true
+        }
+        this.#projectName = bsConfig.projectName
+    }
+
+    async #getBinaryPath(): Promise<string> {
+        if (!this.#binaryPath) {
+            const pb = new PercyBinary()
+            this.#binaryPath = await pb.getBinaryPath(this.#config)
+        }
+        return this.#binaryPath
+    }
+
+    async #sleep(ms: number) {
+        return new Promise(resolve => setTimeout(resolve, ms))
+    }
```

Comment:
duplicate code, let's use a utility function in both classes

Reply:
Done